### PR TITLE
Created Lecture 12 slides

### PR DESCRIPTION
**Pull request by kevincherian**
Lecture 12 slides on model evaluation created.

### CODE REVIEW

**Comment by amueller** [comment]
Thanks. Did you say you'd do slides for this week? It looks like there was a mix-up as I had asked @vinaykale64  to do it on piazza. Sorry if there was any duplicate work :-/

**Comment by kevincherian** [comment]
Yeah, I think I had picked up this week during our meeting. I can talk to Vinay and see if he's already done the other lecture for this week since I haven't started work on that yet!

**Comment by amueller** [comment]
Sorry about the mix-up.
The next one should actually be on imbalanced data, which was 15 before.

**Comment by kevincherian** [comment]
So just to clarify, lectures for next week are Lecture 12 and 15?

**Comment by amueller** [comment]
yes

**Comment by vinaykale64** [comment]
So are the lectures 14, 16 for the next-to-next week?
Just confirming.

**Comment by amueller** [comment]
I'm not sure yet whether I'll do 14 or a case study in the week after. But then there's the midterm on the 7th, and then spring break. But yes, 14 and 16 are next, but 16 will definitely not happen before spring break. It would be good to have 14 ready. I might also do a review for the mid-term.

**Comment by amueller** [comment]
The schedule is on the class website, btw ;)
http://www.cs.columbia.edu/~amueller/comsw4995s18/schedule/

The question is just what I'll do on 03/05/18: case study, mid-term preparation or possibly PCA/ manifold learning.

File: slides/aml-12-022618-model-evaluation/index.html [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+param_grid = {'C': np.logspace(-3, 2, 6),
+              'gamma': np.logspace(-3, 2, 6) / X_train.shape[0]}
+grid = GridSearchCV(SVC(), param_grid=param_grid, cv=10)
+grid.fit(X_train, y_train)
+print(grid.best_params_)
+print(grid.score(X_test, y_test))
+print(len(grid.best_estimator_.support_))
+```
+```
+{'C': 10.0, 'gamma': 0.074239049740163321}
```
Comment (amueller) [inline slides/aml-12-022618-model-evaluation/index.html]: Can you please round all numbers to 3 decimal places?

File: slides/aml-12-022618-model-evaluation/index.html [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+]
+AP only considers ranking!
+???
+---
+
+# ROC Curve
+.center[
+![:scale 100%](images/zoo.png)
+]
+`$$ FPR = \frac{FP}{FP+TN}$$`
```
Comment (amueller) [inline slides/aml-12-022618-model-evaluation/index.html:361]: Can you use ``\text`` for all the acronyms and also make the font bigger for this?

File: slides/aml-12-022618-model-evaluation/index.html [hunk 3]
```diff
@@ -0,0 +1,695 @@
...
+- Strings:
+
+.tiny-code[
+```python
+from sklearn.metrics.scorer import SCORERS
+print("\n".join(sorted(SCORERS.keys())))
+```
+
+```
+accuracy                    precision_macro
```
Comment (amueller) [inline slides/aml-12-022618-model-evaluation/index.html]: you're missing a lot here!

**Comment by amueller** [comment]
looks great apart from minor nitpicks, thanks!

**Comment by amueller** [comment]
Another nitpick: don't sent PRs from master, create a feature branch on your side. (don't change that now, just for the future).

File: slides/aml-12-022618-model-evaluation/index.html (hunk 2 above)
Comment (kevincherian) [reply slides/aml-12-022618-model-evaluation/index.html:361]: Done

File: slides/aml-12-022618-model-evaluation/index.html (hunk 1 above)
Comment (kevincherian) [reply slides/aml-12-022618-model-evaluation/index.html]: Done!

File: slides/aml-12-022618-model-evaluation/index.html (hunk 3 above)
Comment (kevincherian) [reply slides/aml-12-022618-model-evaluation/index.html]: Updated to include remaining scorers!

**Comment by kevincherian** [comment]
Updated with required changes!

**Comment by amueller** [comment]
thanks!

File: slides/aml-12-022618-model-evaluation/index.html [hunk 4]
```diff
@@ -0,0 +1,707 @@
...
+---
+# Multi-class ROC AUC
+- Hand &amp; Till, 2001, one vs one
+`$$ \frac{1}{c(c-1)}\sum_{j=1}^{c}\sum_{k \neq j}^{c} AUC(j,k)$$`
+- Provost &amp; Domingo, 2000, one vs rest
+`$$ \frac{\sum_{j=1}^{c}p(j) AUC(j,{rest}_j)}{c}$$`
+- https://github.com/scikit-learn/scikit-learn/pull/7663
+???
+---
+# Pickling metrics ?
```
Comment (amueller) [inline slides/aml-12-022618-model-evaluation/index.html:480]: pickling? ;) (should be picking)

Comment (kevincherian) [reply slides/aml-12-022618-model-evaluation/index.html:480]: Oops I just saw this, should I make the update now?

Comment (kevincherian) [reply slides/aml-12-022618-model-evaluation/index.html:480]: Looks like the correction has been made!

Comment (amueller) [reply slides/aml-12-022618-model-evaluation/index.html:480]: yeah, no worries, I fixed it.
